test/helpers/helpers.go

```diff
@@ -373,3 +387,104 @@ func CheckCscSuccessfulDeletion(client test.FrameworkClient, cscName, namespace,
 
 	return nil
 }
+
+// RestartMarketplace scales the marketplace deployment down to zero and then scales
+// it back up to it's original number of replicas, and waits for a successful deployment.
+func RestartMarketplace(client test.FrameworkClient, namespace string) error {
+	marketplace := &apps.Deployment{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: "marketplace-operator", Namespace: namespace}, marketplace)
```

Comment:
To address the problem of this test always failing when you run with `operator-sdk up local`, you could make this test contingent on the presence of the `marketplace-operator` deployment.